src/query/atomWithQuery.ts

```diff
@@ -95,6 +94,14 @@ export function atomWithQuery<
     isMounted: boolean
     unsubscribe: (() => void) | null
   }
+
+  let observer: QueryObserver<
```

Comment:
this is going to be a module scope variable, which breaks the use case with multiple `<Provider>`s.

Reply:
Isn't refAtom the best solution for that now?

Reply:
It's better, but not best. I still believe there's a way without refAtom.